examples/PT_afm_heisenberg.jl

```diff
@@ -0,0 +1,37 @@
+using Sunny
+
+# run using the command: "mpiexec -n [n_procs] julia --project [PT_afm_heisenberg.jl]"
+
+#######################################################
+#   system setup
+
+# make lattice
+crystal = Sunny.diamond_crystal(a=1.0)
+
+# interactions -- units of K           
+J = 28.28
+interactions = [
+    heisenberg(J, Bond{3}(3, 6, [0,0,0])),
+]
+
+# spin system
+extent = (4,4,4)
+sys = SpinSystem(crystal, interactions, extent)
+sys_size = length(sys)
+rand!(sys)
+
+#######################################################
+#   PT setup
+
+T_min = 22.5
+T_max = 27.5
+
+# temperature schedule: should take index (i) and number of processes (N) and return temperature (T)
+T_sched(i, N) = (10 .^(range(log10(T_min), log10(T_max), length=N)) )[i]
+
+# make replica for PT
+replica = Replica(MetropolisSampler(sys, 1.0, 1))
```

Comment:
Out of curiosity -- what happens if the user script makes two `Replica`'s? Maybe it should error, if that's detect-able.

Reply:
It still seems to work fine if this happens. I tried it and realized that the spin configurations become de-synchronized with the  local running E, M in the Metropolis sampler if run! is called for each replica. I've added a line to fix this by resetting these quantities both at the start of each run, and after replica exchanges.